### ISSUE TITLE
Authentication with puppeteer or cypress?

### ISSUE THREAD

**Issue by ntucker** [foo-software/lighthouse-check#43]
Lighthouse itself has some guides like https://github.com/GoogleChrome/lighthouse/blob/master/docs/puppeteer.md but it's missing a lot of great utility this library has. Is there a to piggy back a setup like in this guide? If so it would be great to have docs on this

**Comment by adamhenson** [comment]
@ntucker can you rephrase your question in a way that is clearer? What would you specifically like to be documented?

**Comment by ntucker** [comment]
To use this with cypress or puppeteer so authentication (login flows) can happen before the tests run

**Comment by adamhenson** [comment]
Thanks @ntucker - this is out of scope of what this package aims to do. It's really just meant to add extra utility like s3 uploads, GitHub PR comments, etc for basic implementations of Lighthouse. More complex implementations should use Lighthouse directly.

**Comment by ntucker** [comment]
So I have to write S3 uploads, GitHub PR comments myself as this library isn't meant to be used compositionally?

**Comment by adamhenson** [comment]
From what I can understand, the scope of what you need is outside of the scope this project was built for. Your request or problem isn't very clear based in the description and comments, but I'd encourage you to attempt what you're trying to do with this package. If you hit a blocker, you can open an issue that specifically states the problem.

This is an open-source project. PRs that extend and enhance the original scope of this project are certainly welcome.

**Comment by ntucker** [comment]
Well the way they do it in that example is simply allowing a port option

```
const {lhr} = await lighthouse(url, {
  port: (new URL(browser.wsEndpoint())).port,
  output: 'json',
  logLevel: 'info',
});
```

**Comment by adamhenson** [comment]
You can pass in `port` by overriding the default options. You would need to create a config JSON file as [documented here](https://github.com/foo-software/lighthouse-check/#overriding-config-and-option-defaults). Since we're only supporting a JSON file, you might have an issue since in the above example it appears that the port is dynamic.
